src/libmongoc/src/mongoc/mongoc-client-session.c

```diff
@@ -1613,3 +1613,11 @@ _mongoc_client_session_pin (mongoc_client_session_t *session,
 
    session->server_id = server_id;
 }
+
+bool
+mongoc_client_session_get_dirty (mongoc_client_session_t *session)
+{
+   BSON_ASSERT_PARAM (session);
+
+   return session->server_session->dirty;
```

Comment:
Note, no need to assert `server_session` as it is added on construction of a client session. `mongoc_client_session_get_lsid` makes the same assumption.

Reply:
I see. It does look like we should assert the `server_session` arg in `_mongoc_client_session_new` is never `NULL`, though—that way, our assumption that a `mongoc_client_session_t` always has a non-NULL `server_session` is enforced. 

Since that function is only used in by `mongoc_client_start_session`, which checks the server session is non-NULL, it doesn't look like it'd be a problem for now. So I'm happy leaving this as is.

Reply:
I agree, added.